src/components/Wallet/WalletGenerator.jsx

```diff
@@ -80,6 +82,9 @@ class WalletGenerator extends React.Component {
       common: { nodesLoaded },
       setIsWallet,
     } = this.props;
+    if (client.type === "unknown" && prevProps.client.type === "public") {
+      this.setState({ unknownClient: true }); // eslint-disable-line react/no-did-update-set-state
+    }
```

Comment:
I found that the reason the client picker would still show up if you cleared the wallet and then entered a new one is that this component, `WalletGenerator.jsx`, never gets unmounted. It's annoyingly named since it actually mounted for the whole wallet cycle and interaction lifecycle. Since it never unmounts, once this is set to `unknownClient: true` it never gets reset even when you clear the wallet, which only clears out redux. It's unfortunately not possible to clear internal state of components with our `RESET_WALLET` action. Luckily there's a small additional conditional we can add to catch this. 
```js
    const {
      setPasswordError,
      network,
      client,
      common: { nodesLoaded },
      setIsWallet,
      configuring,
    } = this.props;
    const { unknownClient } = this.state;

    if (client.type === "unknown" && prevProps.client.type === "public") {
      this.setState({ unknownClient: true }); // eslint-disable-line react/no-did-update-set-state
    } else if (configuring && client.type !== "unknown" && unknownClient) {
      // re-initializes the state if we're in the configuring stage.
      // catches situation where wallet is cleared and new one is added
      // eslint-disable-next-line react/no-did-update-set-state
      this.setState({ unknownClient: false });
    }
```